packages/lit-dev-discord-bot/src/index.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {
+  ChatInputCommandInteraction,
+  Client,
+  GatewayIntentBits,
+  InteractionType,
+  REST,
+  Routes,
+  SlashCommandBuilder,
+} from 'discord.js';
+import algolia from 'algoliasearch/lite.js';
+import {publicVars} from 'lit-dev-tools-esm/lib/configs.js';
+
+// set up algolia search
+const algClient = algolia(
+  publicVars.algolia.appId,
+  publicVars.algolia.searchOnlyKey
+);
+const index = algClient.initIndex(publicVars.algolia.index);
+
+// The is the GH action secret under the name LIT_DEV_DISCORD_BOT_CLIENT_TOKEN
+const BOT_CLIENT_SECRET = process.env.BOT_CLIENT_SECRET;
+
+if (!BOT_CLIENT_SECRET) {
+  throw new Error('Missing BOT_CLIENT_SECRET');
+}
+
+interface Suggestion {
+  id: number;
+  relativeUrl: string;
+  heading: string;
+  isSubsection: boolean;
+  title: string;
+}
+
+// Build the UI for the slash command.
+const command = new SlashCommandBuilder()
+  .setName('docs')
+  .setDescription('Will search lit.dev.')
+  .addStringOption((option) =>
+    option
+      .setName('query')
+      .setDescription('The query to search for.')
+      .setRequired(true)
+      .setAutocomplete(true)
+  );
+
+const rest = new REST({version: '10'}).setToken(BOT_CLIENT_SECRET);
+
+(async () => {
```

Comment:
Style/readability nit: make this a named function and call it? A name like `publishDiscordCommands()` would be self-documenting.